### PR TITLE
Add a test for path_is_read_only_fs()

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/test/test-stat-util.c [hunk 1]
```diff
@@ -68,15 +69,45 @@ static void test_path_is_fs_type(void) {
...
 static void test_path_is_temporary_fs(void) {
+        const char *s;
+        int r;
+
         log_info("/* %s */", __func__);
 
+        FOREACH_STRING(s, "/", "/run", "/sys", "/sys/", "/proc", "/i-dont-exist", "/var", "/var/lib") {
+                r = path_is_temporary_fs(s);
+
+                log_info_errno(r, "path_is_temporary_fs(\"%s\"): %d, %s",
```
Comment (yuwata) [inline src/test/test-stat-util.c:82]: ~~Maybe, `log_info_errno()` causes assertion if `r == 0`, no?~~

Ah, I forgot that zero error code is allowed in tests.

File: src/test/test-stat-util.c [hunk 2]
```diff
@@ -68,15 +69,45 @@ static void test_path_is_fs_type(void) {
...
+static void test_path_is_read_only_fs(void) {
+        const char *s;
+        int r;
+
+        log_info("/* %s */", __func__);
+
+        FOREACH_STRING(s, "/", "/run", "/sys", "/sys/", "/proc", "/i-dont-exist", "/var", "/var/lib") {
+                r = path_is_read_only_fs(s);
+
+                log_info_errno(r, "path_is_read_only_fs(\"%s\"): %d, %s",
```
Comment (yuwata) [inline src/test/test-stat-util.c:102]: Same here, `log_info()` should be used.

File: src/test/test-stat-util.c [hunk 3]
```diff
@@ -117,7 +119,12 @@ static void test_fd_is_ns(void) {
         assert_se(fd_is_ns(STDERR_FILENO, CLONE_NEWNET) == 0);
         assert_se(fd_is_ns(STDOUT_FILENO, CLONE_NEWNET) == 0);
 
-        assert_se((fd = open("/proc/self/ns/mnt", O_CLOEXEC|O_RDONLY)) >= 0);
+        fd = open("/proc/self/ns/mnt", O_CLOEXEC|O_RDONLY);
+        if (fd < 0) {
+                assert_se(errno == ENOENT);
+                return;
```
Comment (yuwata) [inline src/test/test-stat-util.c:126]: Maybe, it is better to log about skipping several tests.

File: src/test/test-stat-util.c [hunk 4]
```diff
@@ -175,7 +182,11 @@ static void test_device_path_make_canonical_one(const char *path) {
 
         log_debug("> %s", path);
 
-        assert_se(stat(path, &st) >= 0);
+        if (stat(path, &st) < 0) {
+                assert(errno == ENOENT);
+                return;
```
Comment (yuwata) [inline src/test/test-stat-util.c:189]: Also here.

**Comment by yuwata** [comment]
LGTM.

**Comment by keszybz** [comment]
Updated with more logging.

**Comment by keszybz** [comment]
CI is failing left and right, but like in other PRs.

**Pull request merged**